src/main/java/com/twentyonec/ItemsLogger/utils/Storage.java

```diff
@@ -159,44 +147,52 @@ public void setUpTable() {
 		this.update(createTableQuery);
 	}
 
-	private ItemPlayer initializePlayer(ResultSet rs) {
+	private ItemPlayer initializePlayer(final ResultSet rs) {
 
 		try {
-			UUID uuid = UUID.fromString(rs.getString("uuid"));
-			String inv = rs.getString("inventory");
-			String cause = rs.getString("cause");
-			int x = rs.getInt("loc_x");
-			int y = rs.getInt("loc_y");
-			int z = rs.getInt("loc_z");
-			int experience = rs.getInt("experience");
-			Date date = rs.getDate("date");
-			Time time = rs.getTime("time");
+			final UUID uuid = UUID.fromString(rs.getString("uuid"));
+			final String inv = rs.getString("inventory");
+			final String cause = rs.getString("cause");
+			final int x = rs.getInt("loc_x");
+			final int y = rs.getInt("loc_y");
+			final int z = rs.getInt("loc_z");
+			final int experience = rs.getInt("experience");
+			final Date date = rs.getDate("date");
+			final Time time = rs.getTime("time");
 
 			return new ItemPlayer(uuid, inv, cause, x, y, z, experience, date, time);
-		} catch (SQLException e) {
+		} catch (final SQLException e) {
 			e.printStackTrace();
 			return null;
 		}
 	}
 
-	private CompletableFuture<ItemPlayer[]> retrieveLogListAsync(UUID uuid, String date, String cause, int max, int min) {
+	private CompletableFuture<ItemPlayer[]> retrieveLogListAsync(final UUID uuid, final String date, final String cause, final int max, final int min) {
 
 		return CompletableFuture.supplyAsync(() -> {
-			String sql = " SELECT * FROM itemslogger WHERE uuid = '" + uuid + "'";
-
-			if (cause != null) {
-				sql += " AND cause = '" + cause + "'";
+			String sql = " SELECT * FROM itemslogger WHERE uuid = ?";
+			ResultSet rs = null;
+
+			if (date != null && cause != null) {
+				sql += " AND date = ?";
+				sql += " AND cause = ?";
+				sql += " ORDER BY date DESC, time DESC";
+				rs = storage.query(sql, uuid.toString(), date, cause);
+			} else if (date == null && cause != null) {
+				sql += " AND cause = ?";
+				sql += " ORDER BY date DESC, time DESC";
+				rs = storage.query(sql, uuid.toString(), cause);
+			} else if (date != null && cause == null) {
+				sql += " AND date = ?";
+				sql += " ORDER BY date DESC, time DESC";
+				rs = storage.query(sql, uuid.toString(), date);
+			} else if (date == null && cause == null) {
+				sql += " ORDER BY date DESC, time DESC";
+				rs = storage.query(sql, uuid.toString());
 			}
```

Comment:
I think this is a cleaner way of doing what you want here. Variadic parameters are syntactic sugar for an array parameter so `foo(Object... args)` is equivalent to `foo(Object[] args)` and can take an array parameter

```suggestion
			StringBuilder sb = new StringBuilder(sql);
			List<Object> args = new ArrayList<Object>();
			args.add(uuid.toString());

			if (date != null) {
				sb.append(" AND date = ?);
				args.add(date);
			}

			if (cause != null) {
				sb.append(" AND cause = ?);
				args.add(cause);
			}

			sb.append(" ORDER BY date DESC, time DESC");
			rs = storage.query(sql, args.toArray(new Object[0]));
```